pkg/server/config.go

```diff
@@ -368,11 +368,13 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
 		apiHandler = authorization.WithDeepSubjectAccessReview(apiHandler)
 
 		if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
-			apiHandler = tunneler.WithSyncerTunnel(apiHandler)
+			tunneler := tunneler.NewTunneler()
+			apiHandler = tunneler.SyncerTunnelHandler(apiHandler)
 			apiHandler = WithPodSubresourceProxying(
```

Comment:
Why couldn't `WithPodSubresourceProxying` be also a method of the `Tunneller` ?

Reply:
Done